### PR TITLE
feat(RHINENG-13856): Create progress stepper

### PR DESCRIPTION
**Pull request by johnsonm325**
As part of registration assistant refresh, this PR implements the ProgressStepper and some individual step features.

It does not include actual activation key data or content for the selected OS; but it does include the data necessary to step through the stepper.

To test, replace `<Register />` in `Routes.js` with `<NewRegister />` and uncomment the lazy loaded `NewRegister` component in the same file. Run the app and you should see the new content.

Refer to the mocks in the Jira card for specific design.

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export const regAssistantDescription =
+  'The Insights for RHEL registration assistant will guide you through the setup process for the Red Hat Insights client.';
+
+export const dataCollection =
+  "Learn more about Red Hat Insight's data collection and controls";
```
Comment (adonispuente) [inline src/constants.js:5]: Super nitpick, proper grammar is 

`Red Hat Insights'`

File: src/Routes.js [hunk 2]
```diff
@@ -6,6 +6,11 @@ import Loading from './Components/Loading/Loading';
 const Register = lazy(() =>
   import(/* webpackChunkName: "Register" */ './Routes/Register/Register')
 );
+
```
Comment (adonispuente) [inline src/Routes.js:9]: is this intended? Im not sure why we'd do this unless there was a FF

File: src/Routes/Register/NewRegister.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
```
Comment (adonispuente) [inline src/Routes/Register/NewRegister.js:1]: Why make a <NewRegister> component and not just replace the current one all together? Is this gonna be hidden behind a FF? And if so it needs to be slapped in there

File: src/Components/RegistrationProgress/FirstStep.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          >
+            activation key
+            <Icon className="pf-v5-u-ml-xs">
+              <ExternalLinkAltIcon />
+            </Icon>
+          </Text>{' '}
+          to register. An activation key is a pre-shared token that enables
+          authorized users to register and auto-configure systems.
+        </Text>
+        <Text data-testid="select-activation-key">Activation key</Text>
```
Comment (adonispuente) [inline src/Components/RegistrationProgress/FirstStep.js:59]: This needs to be bolded + added a required icon

File: src/Components/RegistrationProgress/FirstStep.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+      </TextContent>
+      <Select
+        isOpen={isOpen}
+        onSelect={(_, optionName) => handleActivationKeySelect(optionName)}
+        selected={selectedKey}
+        toggle={toggle}
+      >
```
Comment (adonispuente) [inline src/Components/RegistrationProgress/FirstStep.js:66]: Missing the 'show activation key' part but that might just be the missing data you refered too.

**Review by adonispuente — CHANGES_REQUESTED** [review]
It looks like this is supposed to be behind a FF according to the ticket, that needs to be created and slapped in here.

The padding is different since we dont render any content for steps until the previous one is completed (not sure if UX is fine with this, should be).

Looks like just missing some api data in different steps. It looks like theres also supposed to be a wizard with these steps along with a nav item change to Registration assistant but theres no epic. Not sure if thats also suppsed to be included in this PR

File: src/Routes.js (hunk 2 above)
Comment (johnsonm325) [reply src/Routes.js:9]: Yeah, this feature is not complete. I'm giving it the malware treatment where I simply create a new component and we'll replace it when the feature is complete.

File: src/Routes/Register/NewRegister.js (hunk 3 above)
Comment (johnsonm325) [reply src/Routes/Register/NewRegister.js:1]: Mainly because this is a completely different design and it's not complete in this PR.

**Review by adonispuente — APPROVED** [review]
LGTM

**Pull request merged**